lib/kennel/models/record.rb

```diff
@@ -136,27 +138,28 @@ def build_json
       end
 
       def build
-        @validation_errors = []
+        @unfiltered_validation_errors = []
         json = nil
 
         begin
           json = build_json
           (id = json.delete(:id)) && json[:id] = id
-          validate_json(json) if validate
+          validate_json(json)
         rescue StandardError
-          if validation_errors.empty?
-            @validation_errors = nil
+          if unfiltered_validation_errors.empty?
+            @unfiltered_validation_errors = nil
             raise PrepareError, safe_tracking_id
           end
         end
 
-        @as_json = json # Only valid if validation_errors.empty?
+        @filtered_validation_errors = filter_validation_errors
+        @as_json = json # Only valid if filtered_validation_errors.empty?
       end
 
       def as_json
         # A courtesy to those tests that still expect as_json to perform validation and raise on error
-        build if @validation_errors.nil?
-        raise Kennel::ValidationError, "#{safe_tracking_id} #{@validation_errors.first}" unless validation_errors.empty?
+        build if @unfiltered_validation_errors.nil?
```

Comment:
this would make more sense ?
```suggestion
        build if @as_json.nil?
```

Reply:
I don't think it makes any difference, but I haven't checked carefully.